python/tests/compat/conftest.py

```diff
@@ -0,0 +1,142 @@
+        """
+        with tempfile.TemporaryDirectory() as dir:
+            df_load_commands = []
+            for df_name, df_value in dfs.items():
+                parquet_file = os.path.join(dir, f"{df_name}.parquet")
+                df_value.to_parquet(parquet_file)
+                df_load_commands.append(f"{df_name} = pd.read_parquet('{parquet_file}')")
+
+            python_commands = [
+                "from arcticdb import Arctic",
+                "import pandas as pd",
+                "import numpy as np",
+                f"ac = Arctic('{self.uri}')"
+            ] + df_load_commands + python_commands
+
+            python_path = os.path.join(dir, "run.py")
+            with open(python_path, "w") as python_file:
+                python_file.write("\n".join(python_commands))
+
+            result = self.venv.execute_python_file(python_path)
+            if result.returncode != 0:
+                raise ErrorInVenv(f"Executing {python_commands} failed with return code {result.returncode}")
+
+    def init_storage(self):
+        self.execute([])
+
+    def create_library(self, lib_name):
+        return VenvLib(self, lib_name, create_if_not_exists=True)
+
+    def get_library(self, lib_name):
+        return VenvLib(self, lib_name, create_if_not_exists=False)
+
+
+class VenvLib:
+    def __init__(self, arctic, lib_name, create_if_not_exists=True):
+        self.arctic = arctic
+        self.lib_name = lib_name
+        if create_if_not_exists:
+            self.create_lib()
+
+    def create_lib(self):
+        self.arctic.execute([f"ac.create_library('{self.lib_name}')"])
+
+    def execute(self, python_commands, dfs={}):
+        python_commands = [
+            f"lib = ac.get_library('{self.lib_name}')",
+            "print(lib._nvs.lib_cfg())" # Remove
+        ] + python_commands
+        return self.arctic.execute(python_commands, dfs)
+
+    def write(self, sym, df):
+        return self.execute([f"lib.write('{sym}', df)"], {"df": df})
+
+    def assert_read(self, sym, df):
+        python_commands = [
+            f"read_df = lib.read('{sym}').data",
+            "pd.testing.assert_frame_equal(read_df, expected_df)"
+        ]
+        return self.execute(python_commands, {"expected_df": df})
+
+
+@pytest.fixture(
+    scope="session",
+    params=("1.6.2", "4.5.0") # TODO: Extend this list
+)
+def old_venv(request):
+    path = os.path.join("./venvs/", request.param)
+    requirements_file = os.path.abspath(os.path.join("tests", "compat", f"requirements-{request.param}.txt"))
+    with Venv(path, requirements_file) as old_venv:
+        yield old_venv
+
+
+@pytest.fixture(
+    params=("lmdb", "s3_bucket_versioning")
+)
+def arctic_uri(request):
+    """
+    arctic_uri uses either lmdb or s3 uri to be used for creating both old and current Arctic instances.
+
+    We use s3_bucket_versioning because it has ssl disabled which allows running tests for older versions like 1.6.2
```

Comment:
I think it would be more robust to define some of your own fixtures for this, than re-using `s3_bucket_versioning`

Reply:
yes, declared new `s3_ssl_disabled`